contracts/MarginCalculator.sol

```diff
@@ -2,11 +2,280 @@
+        }
+        return
+            FixedPointInt256.min(0, _longAmount.sub(_shortAmount)).sub(
+                FixedPointInt256
+                    .min(_longAmount, _shortAmount)
+                    .mul(FixedPointInt256.max(0, _longStrike.sub(_shortStrike)))
+                    .div(_longStrike)
+            );
+    }
+
+    /**
+     * @dev calculate expired net put spread value.
+     *
+     * Formula: net = ( long cash value * long Amount ) - (short cash value * short amount)
+     *
+     * @return net value
+     */
+    function _calculateExpiredPutNetValue(
+        int256 _shortAmount,
+        int256 _longAmount,
+        int256 _shortCashValue,
+        int256 _longCashValue
+    ) internal pure returns (int256) {
+        return (_longCashValue.mul(_longAmount)).sub(_shortCashValue.mul(_shortAmount));
+    }
+
+    /**
+     * @dev calculate expired net call spread value.
+     *                     ( long cash value * long Amount ) - (short cash value * short amount)
+     *  Formula: net =   -------------------------------------------------------------------------
+     *                                               Underlying price
+     * @return net value
+     */
+    function _calculateExpiredCallNetValue(
+        int256 _shortAmount,
+        int256 _longAmount,
+        int256 _shortCashValue,
+        int256 _longCashValue,
+        int256 _underlyingPriceInt
+    ) internal pure returns (int256) {
+        return (_longCashValue.mul(_longAmount)).sub(_shortCashValue.mul(_shortAmount)).div(_underlyingPriceInt);
+    }
+
+    /**
+     * @dev internal function that ensure each asset type & amout array have the same length, and length <= 1;
+     * @param _vault the vault to check.
+     */
+    function _checkAssetCount(Vault memory _vault) internal pure {
+        // For the currect version, check lengths of short, long, ollateral <= 1.
+        require(_vault.shortOtokens.length <= 1, "MarginCalculator: Too many short otokens in the vault.");
+        require(_vault.longOtokens.length <= 1, "MarginCalculator: Too many long otokens in the vault.");
+        require(_vault.collateralAssets.length <= 1, "MarginCalculator: Too many collateral assets in the vault.");
+
+        require(
+            _vault.shortOtokens.length == _vault.shortAmounts.length,
+            "MarginCalculator: Short asset and amount mismatch."
+        );
+        require(
+            _vault.longOtokens.length == _vault.longAmounts.length,
+            "MarginCalculator: Long asset and amount mismatch."
+        );
+        require(
+            _vault.collateralAssets.length == _vault.collateralAmounts.length,
+            "MarginCalculator: Collateral asset and amount mismatch."
+        );
+    }
+
+    /**
+     * @dev internal function that check the long asset is valid for the short asset
+     * @param _vault the vault to check.
+     */
+    function _checkLongAsset(Vault memory _vault) internal view {
```

Comment:
longOtoken? or longAsset? (might be confusing as it might be confused for collateral)